recipes/newrelic/apm/dotNet/windows/net-install.yml

```diff
@@ -61,7 +61,51 @@ install:
     remove_any_previous:
       ignore_error: true
       cmds:
-        - powershell -command '$app = Get-WmiObject -Class Win32_Product -Filter "Name like 'New Relic .NET Agent%'"; if ($app) { $app.Uninstall() }'
+        - |
+          powershell -command '
+          function Find-UninstallCommand {
+            param (
+              [Parameter(Mandatory)]
+              [string]$Match
+            )
+
+            $baseKeys = Get-ChildItem -Path HKLM:\SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall `
+            | % { $_.Name.TrimStart("HKEY_LOCAL_MACHINE\") }
+
+            $wowKeys = Get-ChildItem -Path HKLM:\SOFTWARE\Wow6432Node\Microsoft\Windows\CurrentVersion\Uninstall `
+            | % { $_.Name.TrimStart("HKEY_LOCAL_MACHINE\") }
+
+            $allKeys = $baseKeys + $wowKeys
+
+            $uninstallIds = New-Object System.Collections.ArrayList
+            foreach ($key in $allKeys) {
+              $keyData = Get-Item -Path HKLM:\$key
+              $name = $keyData.GetValue("DisplayName")
+              if ($name -and $name -match $Match) {
+                $keyId = Split-Path $key -Leaf
+                $uninstallIds.Add($keyId) | Out-Null
+              }
+            }
+
+            if ($uninstallIds.Count -gt 1) {
```

Comment:
This is a little puzzling. I would like to find a way to proceed with the installation without surfacing an error. Could we return the first one that matched and just un-install that one?